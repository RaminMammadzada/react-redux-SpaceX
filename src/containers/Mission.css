.mission-logo-width {
  width: 220px;
}

.mission-link:hover {
  border: 0.2rem rgb(85, 85, 85);
  border-radius: 20%;
  border-style: dashed;
  color: brown;
  background-color: rgb(85, 85, 85, 0.2);
}
