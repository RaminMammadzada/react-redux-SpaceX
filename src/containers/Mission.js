import React from 'react';
import PropTypes from 'prop-types';

const Mission = (props) => {
  const {
    missionImage,
    missionName,
    launchYear,
    launchSuccess,
  } = props;

  return (
    <div>
      <img src={missionImage} alt={missionName} />
      <h5>{missionName}</h5>
      <h5>{launchYear}</h5>
      <h5>{launchSuccess ? 'YES' : 'NO'}</h5>
    </div>
  );
};

Mission.propTypes = {
  missionImage: PropTypes.string,
  missionName: PropTypes.string,
  launchYear: PropTypes.string.isRequired,
  launchSuccess: PropTypes.bool,
};

Mission.defaultProps = {
  missionImage: 'https:',
  missionName: 'name',
  launchSuccess: true,
};

export default Mission;
