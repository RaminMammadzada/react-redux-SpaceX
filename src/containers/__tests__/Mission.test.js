import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store';
import Mission from '../MissionList';

describe('Mission container - ', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Mission
            key="sample_key_a"
            missionImage="sample_mission_image_a"
            missionName="sample_mission_name_a"
            launchYear="sample_mission_launch_year_a"
            launchSuccess="sample_mission_launch_success_a"
            missionId="sample_mission_id_a"
          />
        </Provider>
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
