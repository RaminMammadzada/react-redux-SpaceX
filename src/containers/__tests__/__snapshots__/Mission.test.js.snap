// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission container -  matches the snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="d-flex flex-column align-items-center"
    >
      <h1>
        Launch Missions
      </h1>
      <div>
        <form
          className="d-flex flex-wrap justify-content-center ml-5"
        >
          <div
            className="d-flex align-items-center flex-row mx-5 form-group"
          >
            <label
              className="pr-3 form-label"
              htmlFor="exampleForm.SelectCustom"
            >
              Year
            </label>
            <select
              className="custom-select"
              id="exampleForm.SelectCustom"
              name="year"
              onChange={[Function]}
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value={2000}
              >
                2000
              </option>
              <option
                value={2001}
              >
                2001
              </option>
              <option
                value={2002}
              >
                2002
              </option>
              <option
                value={2003}
              >
                2003
              </option>
              <option
                value={2004}
              >
                2004
              </option>
              <option
                value={2005}
              >
                2005
              </option>
              <option
                value={2006}
              >
                2006
              </option>
              <option
                value={2007}
              >
                2007
              </option>
              <option
                value={2008}
              >
                2008
              </option>
              <option
                value={2009}
              >
                2009
              </option>
              <option
                value={2010}
              >
                2010
              </option>
              <option
                value={2011}
              >
                2011
              </option>
              <option
                value={2012}
              >
                2012
              </option>
              <option
                value={2013}
              >
                2013
              </option>
              <option
                value={2014}
              >
                2014
              </option>
              <option
                value={2015}
              >
                2015
              </option>
              <option
                value={2016}
              >
                2016
              </option>
              <option
                value={2017}
              >
                2017
              </option>
              <option
                value={2018}
              >
                2018
              </option>
              <option
                value={2019}
              >
                2019
              </option>
              <option
                value={2020}
              >
                2020
              </option>
              <option
                value={2021}
              >
                2021
              </option>
            </select>
          </div>
          <div
            className="d-flex align-items-center flex-row mx-5 form-group"
          >
            <label
              className="pr-3 form-label"
              htmlFor="exampleForm.SelectCustom"
            >
              Lauch success
            </label>
            <select
              className="custom-select"
              id="exampleForm.SelectCustom"
              name="successCase"
              onChange={[Function]}
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value="Yes"
              >
                Yes
              </option>
              <option
                value="No"
              >
                No
              </option>
            </select>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="d-flex flex-wrap justify-content-center"
  />
</div>
`;

exports[`Mission container-  matches the snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="d-flex flex-column align-items-center"
    >
      <h1>
        Launch Missions
      </h1>
      <div>
        <form
          className="d-flex flex-wrap justify-content-center ml-5"
        >
          <div
            className="d-flex align-items-center flex-row mx-5 form-group"
          >
            <label
              className="pr-3 form-label"
              htmlFor="exampleForm.SelectCustom"
            >
              Year
            </label>
            <select
              className="custom-select"
              id="exampleForm.SelectCustom"
              name="year"
              onChange={[Function]}
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value={2000}
              >
                2000
              </option>
              <option
                value={2001}
              >
                2001
              </option>
              <option
                value={2002}
              >
                2002
              </option>
              <option
                value={2003}
              >
                2003
              </option>
              <option
                value={2004}
              >
                2004
              </option>
              <option
                value={2005}
              >
                2005
              </option>
              <option
                value={2006}
              >
                2006
              </option>
              <option
                value={2007}
              >
                2007
              </option>
              <option
                value={2008}
              >
                2008
              </option>
              <option
                value={2009}
              >
                2009
              </option>
              <option
                value={2010}
              >
                2010
              </option>
              <option
                value={2011}
              >
                2011
              </option>
              <option
                value={2012}
              >
                2012
              </option>
              <option
                value={2013}
              >
                2013
              </option>
              <option
                value={2014}
              >
                2014
              </option>
              <option
                value={2015}
              >
                2015
              </option>
              <option
                value={2016}
              >
                2016
              </option>
              <option
                value={2017}
              >
                2017
              </option>
              <option
                value={2018}
              >
                2018
              </option>
              <option
                value={2019}
              >
                2019
              </option>
              <option
                value={2020}
              >
                2020
              </option>
              <option
                value={2021}
              >
                2021
              </option>
            </select>
          </div>
          <div
            className="d-flex align-items-center flex-row mx-5 form-group"
          >
            <label
              className="pr-3 form-label"
              htmlFor="exampleForm.SelectCustom"
            >
              Lauch success
            </label>
            <select
              className="custom-select"
              id="exampleForm.SelectCustom"
              name="successCase"
              onChange={[Function]}
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value="Yes"
              >
                Yes
              </option>
              <option
                value="No"
              >
                No
              </option>
            </select>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="d-flex flex-wrap justify-content-center"
  />
</div>
`;
